src/cosl/coordinated_workers/interface.py

```diff
@@ -523,7 +523,16 @@ def publish_app_roles(self, roles: Iterable[str]):
         relation = self.relation
         if relation:
             databag_model = ClusterRequirerAppData(role=",".join(roles))
-            databag_model.dump(relation.data[self.model.app])
+            try:
+                databag_model.dump(relation.data[self.model.app])
+            except ModelError as e:
+                if "ERROR permission denied (unauthorized access)" in e.args:
+                    # if we are handling an event prior to 'start', we could be denied write access
+                    # hopefully we can ignore it and trust that it will be retried at the next occasion
+                    log.debug(
+                        "databag dump gave a permission denied error. "
+                        "This could be a transient issue."
+                    )
```

Comment:
If we think swallowing this error is a good idea, 
- should it be done inside `.dump()`?
- substring check is brittle - I think we can safely log a `debug` for any ModelError.

But I think that this approach may bite us: model errors are not meant to be dropped - worst case we'd defer the event, but without context at hand I think we should opt for one of the following:
- Caller must guard.
- It's ok to go to error state.

Unless I'm missing something?

Reply:
I like the idea of letting the caller guard, so the Worker will have to  do that in `_update_cluster_relation`.
afraid we can't make the substring check any less brittle as that is all we get from ops.

as for the error state, that should be only if we get a different error, one we don't trust is transient.
Updated.